juriscraper/opinions/united_states/state/sc.py

```diff
@@ -10,107 +10,103 @@
                is to ensure that we get a cookie in our session by visiting the
                homepage before we go and scrape. Dumb.
  - 2019-02-26: Restructured completely by arderyp
+ - 2024-09-18: Update to OpinionSiteLinear, implement backscraper,
+    support unpublished opinions, by @grossir
 
 Contact information:
  - Help desk: (803) 734-1193, Travis
  - Security desk: 803-734-5906, Joe Hilke
  - Web Developer (who can help with firewall issues): 803-734-0373, Winkie Clark
 """
 
-import datetime
+from datetime import date
+from typing import Dict, List, Tuple
 
-from juriscraper.lib.exceptions import InsanityException
-from juriscraper.lib.string_utils import convert_date_string
-from juriscraper.OpinionSite import OpinionSite
+from juriscraper.AbstractSite import logger
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
+    # Full URL example:
+    # https://www.sccourts.org/opinions-orders/opinions/published-opinions/supreme-court/?term=2024-09
+    base_url = (
+        "https://www.sccourts.org/opinions-orders/opinions/{}/{}/?term={}-{}"
+    )
+    opinion_status = "published-opinions"
+    court = "supreme-court"
+    days_interval = 27  # guarantees picking each month
+    first_opinion_date = date(2000, 1, 1)
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
-        today = datetime.date.today()
-        self.url = (
-            "http://www.sccourts.org/opinions/indexSCPub.cfm?year=%d&month=%d"
-            % (today.year, today.month)
-        )
-        self.cases = []
+        self.make_backscrape_iterable(kwargs)
+        self.url = self.make_url_from_date(date.today())
+        if self.opinion_status == "published-opinions":
+            self.status = "Published"
+        else:
+            self.status = "Unpublished"
```

Comment:
why not just assign self.status instead of this self.opinion_status

Reply:
self.opinion_status is used to build the URL too 

`self.opinion_status == "published-opinions"` 
https://www.sccourts.org/opinions-orders/opinions/published-opinions/supreme-court/?term=2024-09

`self.opinion_status == "unpublished-opinions"` 
https://www.sccourts.org/opinions-orders/opinions/unpublished-opinions/supreme-court/?term=2024-09